src/Admin/Filter/AdherentTagFilter.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Admin\Filter;
+
+use Sonata\AdminBundle\Filter\Model\FilterData;
+use Sonata\AdminBundle\Form\Type\Operator\ContainsOperatorType;
+use Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQuery;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+class AdherentTagFilter extends AbstractCallbackDecoratorFilter
+{
+    protected function getInitialFilterOptions(): array
+    {
+        return [
+            'field_type' => ChoiceType::class,
+            'operator_type' => ContainsOperatorType::class,
+            'callback' => function (ProxyQuery $qb, string $alias, string $field, FilterData $value) {
+                if (!$value->hasValue()) {
+                    return false;
+                }
+
+                $orX = $qb->expr()->orX();
+
+                $condition = match ($value->getType()) {
+                    ContainsOperatorType::TYPE_NOT_CONTAINS => 'NOT LIKE',
+                    default => 'LIKE',
+                };
+
+                foreach ($value->getValue() as $index => $choice) {
+                    $orX->add($alias.'.tags '.$condition.' :tag_'.$index);
+                    $qb->setParameter('tag_'.$index, '%'.$choice.'%');
```

Comment:
as we can use this filter many times on the same Admin we need customize `:tag_1` query params to be unique in the query